src/Persistence_matrix/include/gudhi/Persistence_matrix/columns/unordered_set_column.h

```diff
@@ -0,0 +1,1023 @@
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Hannah Schreiber
+ *
+ *    Copyright (C) 2022-24 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+/**
+ * @file unordered_set_column.h
+ * @author Hannah Schreiber
+ * @brief Contains the @ref Unordered_set_column class.
+ * Also defines the std::hash method for @ref Unordered_set_column.
+ */
+
+#ifndef PM_UNORDERED_SET_COLUMN_H
+#define PM_UNORDERED_SET_COLUMN_H
+
+#include <vector>
+#include <stdexcept>
+#include <type_traits>
+// #include <unordered_set>
+#if BOOST_VERSION >= 108100
+#include <boost/unordered/unordered_flat_set.hpp>
+#else
+#include <boost/unordered_set.hpp>  // preferably with boost 1.79+ for speed
+#endif
+#include <set>
+#include <utility>  //std::swap, std::move & std::exchange
+
+#include <boost/iterator/indirect_iterator.hpp>
+
+#include <gudhi/Persistence_matrix/allocators/cell_constructors.h>
+
+namespace Gudhi {
+namespace persistence_matrix {
+
+/**
+ * @class Unordered_set_column unordered_set_column.h gudhi/Persistence_matrix/columns/unordered_set_column.h
+ * @ingroup persistence_matrix
+ *
+ * @brief Column class following the @ref PersistenceMatrixColumn concept.
+ *
+ * Column based on an unordered set structure. The cells are not ordered, but only non-zero values
+ * are stored uniquely in the underlying container.
+ * 
+ * @tparam Master_matrix An instanciation of @ref Matrix from which all types and options are deduced.
+ * @tparam Cell_constructor Factory of @ref Cell classes.
+ */
+template <class Master_matrix>
+class Unordered_set_column : public Master_matrix::Row_access_option,
+                             public Master_matrix::Column_dimension_option,
+                             public Master_matrix::Chain_column_option 
+{
+ public:
+  using Master = Master_matrix;
+  using index = typename Master_matrix::index;
+  using id_index = typename Master_matrix::id_index;
+  using dimension_type = typename Master_matrix::dimension_type;
+  using Field_element_type = typename Master_matrix::element_type;
+  using Cell = typename Master_matrix::Cell_type;
+  using Column_settings = typename Master_matrix::Column_settings;
+
+ private:
+  using Field_operators = typename Master_matrix::Field_operators;
+  using Cell_constructor = typename Master_matrix::Cell_constructor;
+
+  struct CellPointerHash {
+    size_t operator()(const Cell* c) const { return std::hash<Cell>()(*c); }
+  };
+  struct CellPointerEq {
+    bool operator()(const Cell* c1, const Cell* c2) const { return *c1 == *c2; }
+  };
+  struct CellPointerComp {
+    bool operator()(const Cell* c1, const Cell* c2) const { return *c1 < *c2; }
+  };
+
+#if BOOST_VERSION >= 108100
+  using Column_type = boost::unordered_flat_set<Cell*, CellPointerHash, CellPointerEq>;
+#else
+  using Column_type = boost::unordered_set<Cell*, CellPointerHash, CellPointerEq>;
+#endif
```

Comment:
Did you have a chance to check the impact on performance? Since you iterate on the elements on the map (whereas the "normal" operations are only find/insert/erase), it isn't completely obvious if that helps or hurts (I would bet a bit on "help").

Reply:
As I mentioned somewhere else, the times with `unordered_flat_set` are much better. But I tested everything with the same boost version (the latest). And there is the comment "preferably with boost 1.79+ for speed" beside `#include <boost/unordered_set.hpp>` in the example you gave me. For Boost 1.78 and less, would it be preferable to use the standard `std::unordered_set` as it could potentially be more up to date than the one from an old boost version?

Reply:
Let's not care about those old boost versions.